### PR TITLE
SRCH-3654, SRCH-3993, SRCH-3657: search page alert integration, spelling suggestion UI and integration

### PR DESCRIPTION
**Pull request by yogesh27**
## Summary
- SRCH-3654: search page alert integration
- SRCH-3993: spelling suggestion UI
- SRCH-3657: spelling  suggestion integration
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
  
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [x] Automated checks pass. If Code Climate checks do not pass, explain reason for failures:
 
#### Process Checks

- [x] You have specified at least one "Reviewer".

### CODE REVIEW

File: features/redesigned_searches.feature [hunk 1]
```diff
@@ -393,3 +393,29 @@ Feature: Search - redesign
 
     When I follow "1"
     Then I should see exactly "20" redesigned video search results
+
+  @javascript @a11y
+  Scenario: Display an Alert on search page
+    Given the following Affiliates exist:
+      | display_name | name          | contact_email    | first_name | last_name | locale              |  use_redesigned_results_page |
```
Comment (stevenbarragan) [inline features/redesigned_searches.feature:400]: please fix alignment issue

File: features/redesigned_searches.feature [hunk 2]
```diff
@@ -393,3 +393,29 @@ Feature: Search - redesign
...
+    When I am on en.agency.gov's search page
+    Then I should not see "New alert for the test aff"
+
+  @javascript @a11y @a11y_wip
+  Scenario: Searching with spelling suggestions
+    Given the following Affiliates exist:
+      | display_name | name       | contact_email | first_name | last_name | domains | use_redesigned_results_page |
+      | agency site  | agency.gov | aff@bar.gov   | Jane       | Bar       | usa.gov | true |
+    When I am on agency.gov's search page
+    And I search for "qeury" in the redesigned search page
```
Comment (stevenbarragan) [inline features/redesigned_searches.feature:419]: typo? "qeury" => "query"

File: features/redesigned_searches.feature [hunk 3]
```diff
@@ -393,3 +393,29 @@ Feature: Search - redesign
...
+    Then I should see "Showing results for query"
+    And I should see "Search instead for qeury"
```
Comment (stevenbarragan) [inline features/redesigned_searches.feature:421]: Is this the spelling suggestion?

If it is I'm not sure where it's coming from and it's a little confusing that you search for "qeury" and then you get "Search instead for qeury". It seems that something is missing here.

**Review by stevenbarragan — CHANGES_REQUESTED** [review]

File: features/redesigned_searches.feature (hunk 3 above)
Comment (yogesh27) [reply features/redesigned_searches.feature:421]: this is for spelling suggestion.

File: features/redesigned_searches.feature (hunk 2 above)
Comment (yogesh27) [reply features/redesigned_searches.feature:419]: this is for spelling suggestion.

File: features/redesigned_searches.feature (hunk 1 above)
Comment (yogesh27) [reply features/redesigned_searches.feature]: fixed.

**Review by stevenbarragan — CHANGES_REQUESTED** [review]

File: features/redesigned_searches.feature [hunk 4]
```diff
@@ -393,3 +393,29 @@ Feature: Search - redesign
...
+    Given the following Alert exists:
+      | affiliate    | text                       | status   | title        |
+      | en.agency.gov| New alert for the test aff | Active   |  Test Title  |
+    When I am on en.agency.gov's search page
+    Then I should see "New alert for the test aff"
+    Given the following Alert exists:
+      | affiliate    | text                       | status   | title      |
+      | en.agency.gov| New alert for the test aff | Inactive | Test Title |
+    When I am on en.agency.gov's search page
+    Then I should not see "New alert for the test aff"
```
[the remark this answers is not in the document]
Comment (yogesh27) [reply features/redesigned_searches.feature:411]: @stevenbarragan I carried over the same test from here: https://github.com/GSA/search-gov/blob/main/features/searchgov_search.feature#L45

[the remark this answers is not in the document]
Comment (yogesh27) [reply features/redesigned_searches.feature:417]: I have updated the scenarios

**Review by stevenbarragan — APPROVED** [review]

**Pull request merged**